### PR TITLE
[AKS] Add ```--disable-windows-outbound-nat``` for ```az aks nodepool add``` to add a Windows agent pool which the Windows OutboundNAT is disabled

### PR DESCRIPTION
**Pull request by ShiqianTao**
[AKS] Add ```--disable-windows-outbound-nat``` for ```az aks nodepool add``` to add a Windows agent pool which the Windows OutboundNAT is disabled

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

```
az aks nodepool add --name npwin --os-type Windows --disable-windows-outbound-nat
```

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by ShiqianTao** [comment]
Hi @FumingZhang @AbelHu , could you help to review this PR? Thank you.

File: src/aks-preview/azext_aks_preview/_validators.py [hunk 1]
```diff
@@ -639,6 +639,14 @@ def validate_enable_custom_ca_trust(namespace):
                 '--enable_custom_ca_trust can only be set for Linux nodepools')
 
 
+def validate_disable_windows_outbound_nat(namespace):
+    """Validates disable_windows_outbound_nat can only be used on Windows."""
+    if namespace.disable_windows_outbound_nat:
+        if hasattr(namespace, 'os_type') and namespace.os_type != "Windows":
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_validators.py:645]: Should handle case-insensitive comparation?

File: src/aks-preview/azext_aks_preview/agentpool_decorator.py [hunk 2]
```diff
@@ -328,6 +372,8 @@ def construct_agentpool_profile_preview(self) -> AgentPool:
         agentpool = self.set_up_gpu_properties(agentpool)
         # set up custom ca trust
         agentpool = self.set_up_custom_ca_trust(agentpool)
+        # set up custom ca trust
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/agentpool_decorator.py:375]: misleading comment

File: src/aks-preview/azext_aks_preview/agentpool_decorator.py [hunk 3]
```diff
@@ -309,6 +336,23 @@ def set_up_custom_ca_trust(self, agentpool: AgentPool) -> AgentPool:
...
+        """Set up windows profile for the AgentPool object.
+
+        :return: the AgentPool object
+        """
+        self._ensure_agentpool(agentpool)
+
+        disable_windows_outbound_nat = self.context.get_disable_windows_outbound_nat()
+
+        # Construct AgentPoolWindowsProfile if one of the fields has been set
+        if disable_windows_outbound_nat:
```
Comment (AbelHu) [inline src/aks-preview/azext_aks_preview/agentpool_decorator.py:349]: does it mean that we will ignore the value when the user sets it to false? If this is only planned to be used in agent pool creation, it is fine. If this function will be reused in agent pool update operation, we may need to change it.

**Comment by FumingZhang** [comment]
Queued a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=62164527&view=results) to test the newly added case.

File: src/aks-preview/azext_aks_preview/agentpool_decorator.py (hunk 3 above)
Comment (ShiqianTao) [reply src/aks-preview/azext_aks_preview/agentpool_decorator.py:349]: According to my reading from the history code, we should add a new function if we support the update operation:
1. Update opration's functions in the class AKSPreviewAgentPoolUpdateDecorator - https://github.com/Azure/azure-cli-extensions/blob/7962586a9d704be07d71be44f18c545550ef9e46/src/aks-preview/azext_aks_preview/agentpool_decorator.py#L337
1. Usually, the CLI usually exposes two options for boolean selection, e.g., ```disable-windows-outbound-nat``` and ```enable-windows-outbound-nat```. If we add support for updating the existing agent pool, we should always add support on CLI.

File: src/aks-preview/azext_aks_preview/_validators.py (hunk 1 above)
Comment (ShiqianTao) [reply src/aks-preview/azext_aks_preview/_validators.py]: Improved. Thank you.

File: src/aks-preview/azext_aks_preview/agentpool_decorator.py (hunk 2 above)
Comment (ShiqianTao) [reply src/aks-preview/azext_aks_preview/agentpool_decorator.py]: Updated. Thank you.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by AbelHu — APPROVED** [review]

**Comment by ShiqianTao** [comment]
Hi @zhoxing-ms , cloud you help to review and approve this PR? Thank you.

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**